src/stavesection.ts

```diff
@@ -51,32 +52,32 @@ export class StaveSection extends StaveModifier {
   }
 
   draw(stave: Stave, shift_x: number): this {
+    const borderWidth = 2;
+    const padding = 2;
     const ctx = stave.checkContext();
     this.setRendered();
 
     ctx.save();
-    ctx.setLineWidth(2);
+    ctx.setLineWidth(borderWidth);
     ctx.setFont(this.textFont);
+    const textFormatter = TextFormatter.create(this.textFont);
 
-    const paddingX = 2;
-    const paddingY = 2;
-    const rectWidth = 2;
-    const textMeasurements = ctx.measureText(this.section);
-    const textWidth = textMeasurements.width;
-    const textHeight = textMeasurements.height;
-    const width = textWidth + 2 * paddingX; // add left & right padding
-    const height = textHeight + 2 * paddingY; // add top & bottom padding
+    const textWidth = textFormatter.getWidthForTextInPx(this.section);
+    const textY = textFormatter.getYForStringInPx(this.section);
+    const textHeight = textY.height;
+    const headroom = -1 * textY.yMin;
+    const width = textWidth + 2 * padding; // add left & right padding
+    const height = textHeight + 2 * padding; // add top & bottom padding
 
     //  Seems to be a good default y
     const y = stave.getYForTopText(1.5) + this.shift_y;
     const x = this.x + shift_x;
     if (this.drawRect) {
       ctx.beginPath();
-      ctx.setLineWidth(rectWidth);
-      ctx.rect(x, y + textMeasurements.y - paddingY, width, height);
+      ctx.rect(x, y - height + headroom, width, height);
```

Comment:
line width is no longer  set to 2

Reply:
I mean that the call to setLineWidth is no longer there, is that on purpose?

Reply:
Oh...right, because it's called on line 61.  This call was redundant.